redacted-compiler-plugin/src/main/kotlin/dev/zacsweers/redacted/compiler/ir/RedactedIrVisitor.kt

```diff
@@ -205,7 +207,7 @@ internal class RedactedIrVisitor(
   }
 
   private fun IrClass.reportError(message: String) {
-    val location = CompilerMessageLocation.create(name.asString())
+    val location = MessageUtil.psiElementToMessageLocation(descriptor.source.getPsi())
```

Comment:
Is there no way to do this without the deprecated descriptor APIs?

Reply:
Oh shoot, looks like there [might be](https://github.com/JetBrains/kotlin/blob/29b23e79f32791e456a5b4a453277f0f0b3e984d/compiler/ir/ir.psi2ir/src/org/jetbrains/kotlin/psi2ir/PsiSourceManager.kt#L141-L142)! I didn't look hard enough.

Reply:
It doesn't look like a populated `PsiSourceManager` ever makes it down to us though. `GeneratorContext` has it.

Reply:
Private APIs _shakes fist_

Reply:
Filed a request since this keeps coming up https://youtrack.jetbrains.com/issue/KT-44277